### PR TITLE
feat: iceberg table engine.

### PR DESCRIPTION
**Pull request by youngsofun**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

part of https://github.com/datafuselabs/databend/issues/13787

 use `table options` 

`create table <name> engine = iceberg location=<location> [connection_name =<connection_name>`]

- - -
This change is [Reviewable](https://reviewable.io/reviews/datafuselabs/databend/13835)

### CODE REVIEW

File: tests/suites/1_stateful/10_iceberg/10_0002_iceberg_engine.result [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+3	c
+4	d
+5	e
+6	d
+<<<<
+>>>> show create table test_iceberg;
+test_iceberg	CREATE TABLE `test_iceberg` (
+  `id` INT NOT NULL,
+  `data` VARCHAR NOT NULL
+) ENGINE=ICEBERG
```
Comment (sundy-li) [inline tests/suites/1_stateful/10_iceberg/10_0002_iceberg_engine.result:15]: seems  the location is missing in show stmt

Comment (youngsofun) [reply tests/suites/1_stateful/10_iceberg/10_0002_iceberg_engine.result:15]: I think there is the same for `create table with external location`:

```
pub struct ShowCreateTablePlan {
    /// The catalog name
    pub catalog: String,
    /// The database name
    pub database: String,
    /// The table name
    pub table: String,
    /// The table schema
    pub schema: DataSchemaRef,
}
```

----

the problem is    `<url> [connection=]`  is translated  to `stroage parameters`.
need a way to restore it? maybe not always feasible.

Comment (Xuanwo) [reply tests/suites/1_stateful/10_iceberg/10_0002_iceberg_engine.result:15]: > the problem is `<url> [connection=]` is translated to `stroage parameters`.
> need a way to restore it?

Yep, we are not ready for transform stroage parameters into url yet.

**Comment by youngsofun** [comment]
another issue:

in current impl of this pr.
url + connenction is translated to storage_parameter.
if url is a stage (object), or connection contain named connection object.
the location for table should change with the  stage/connection object.

so I now prefer use table options:

`create table <name> engine = iceberg location=<location> connection=<connection_name>`

**Comment by youngsofun** [comment]
> another issue:
> 
> 
> 
> in current impl of this pr.
> 
> url + connenction is translated to storage_parameter.
> 
> if url is a stage (object), or connection contain named connection object.
> 
> the location for table should change with the  stage/connection object.
> 
> 
> 
> so I now prefer use table options:
> 
> 
> 
> `create table <name> engine = iceberg location=<location> connection=<connection_name>`
> 
> cc @Xuanwo @BohuTANG 
> 
> 
> 
> acturally the gramma for ending `<url> [connection=]` is a little strange to me.
> 
> 
> 
> 
> 
> 
> 
> 
in this way， we share the same grammar for options with table fuctions like inferschema

File: tests/suites/1_stateful/10_iceberg/10_0002_iceberg_engine.result (hunk 1 above)
Comment (youngsofun) [reply tests/suites/1_stateful/10_iceberg/10_0002_iceberg_engine.result]: > seems the location is missing in show stmt

fixed

**Comment by youngsofun** [comment]
```
  ==> installing ziglang...
  Installing zig.
  Error: Not tolerating missing names on install, stopping.
  Error: The process '/usr/local/bin/docker' failed with exit code 1
      at ExecState._setResult (/runner/_work/_actions/PyO3/maturin-action/v1/dist/index.js:1702:25)
      at ExecState.CheckComplete (/runner/_work/_actions/PyO3/maturin-action/v1/dist/index.js:1685:18)
      at ChildProcess.<anonymous> (/runner/_work/_actions/PyO3/maturin-action/v1/dist/index.js:1579:27)
      at ChildProcess.emit (node:events:513:28)
      at maybeClose (node:internal/child_process:1100:16)
      at Process.ChildProcess._handle.onexit (node:internal/child_process:304:5)
  Error: The process '/usr/local/bin/docker' failed with exit code 1
```

cc @everpcpc

**Comment by youngsofun** [comment]
@Xuanwo @sundy-li 

please review again

**Review by Xuanwo — APPROVED** [review]

**Pull request merged**